libcloud/compute/drivers/gce.py

```diff
@@ -1587,6 +1619,43 @@ def ex_list_routes(self):
                        response.get('items', [])]
         return list_routes
 
+    def ex_list_subnetworks(self, region=None):
+        """
+        Return the list of subnetworks.
+
+        :keyword  region: Region for the subnetwork. Specify 'all' to return
+                          the aggregated list of subnetworks.
+        :type     region: ``str`` or :class:`GCERegion`
+
+        :return: A list of subnetwork objects.
+        :rtype: ``list`` of :class:`GCESubNetwork`
+        """
+        region = self._set_region(region)
+        if region is None:
+            request = '/aggregated/subnetworks'
+        else:
+            request = '/regions/%s/subnetworks' % (region.name)
+
+        list_subnetworks = []
+        response = self.connection.request(request, method='GET').object
+
+        if 'items' in response:
+            if region is None:
+                for v in response['items'].values():
```

Comment:
That's quite some levels of nesting, would probably make it a bit easier to rid if this functionality was refactored in a separate method.

In any case, not a blocker for me.